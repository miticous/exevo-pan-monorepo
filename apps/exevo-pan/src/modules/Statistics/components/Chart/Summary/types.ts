export interface SummaryProps extends React.HTMLAttributes<HTMLDivElement> {
  title: string
  value: number
  percentage: number
  positive?: boolean
}

export interface PositiveStyleProps {
  positive: boolean
}
