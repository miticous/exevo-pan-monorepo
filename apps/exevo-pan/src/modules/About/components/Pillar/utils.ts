export const generateNavId = (id?: string): string => `${id}-nav`
