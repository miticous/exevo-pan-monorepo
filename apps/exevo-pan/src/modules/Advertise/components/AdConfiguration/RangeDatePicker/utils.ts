const formatDate = (timestamp: number): string => {
  const date = new Date(timestamp)
  const day = date.getDate().toString()
  const month = (date.getMonth() + 1).toString()
  const year = date.getFullYear().toString()

  return `${month}/${day}/${year}`
}

export const getDaysUntilAuctionEnd = (auctionEnd?: number): string[] => {
  if (!auctionEnd) return []

  const formatedEndDate = formatDate(auctionEnd * 1000)
  const formatedCurrentDate = formatDate(+new Date())

  const endDate = new Date(formatedEndDate)
  const currentDate = new Date(formatedCurrentDate)
  const daysInRange = [formatedCurrentDate]
  while (currentDate < endDate) {
    currentDate.setDate(currentDate.getDate() + 1)
    daysInRange.push(formatDate(+currentDate))
  }

  return daysInRange
}

export const partitionByMonths = (dateArray: string[]): string[][] => {
  const monthsPartition = new Set<string>([])

  dateArray.forEach((dateString) => {
    const [month] = dateString.split('/')
    monthsPartition.add(month)
  })

  return [...monthsPartition].map((month) =>
    dateArray.filter((dateString) => {
      const [dateStringMonth] = dateString.split('/')

      return month === dateStringMonth
    }),
  )
}

export const fillWithDays = (
  initialDateString: string,
  amount: number,
  step: number,
): string[] => {
  const currentDate = new Date(initialDateString)

  return Array.from(
    { length: amount },
    () => +currentDate.setDate(currentDate.getDate() + step),
  )
    .sort((a, b) => a - b)
    .map(formatDate)
}

export const getWeekdayNumber = (date: string): number =>
  new Date(date).getDay()

export const getDay = (date: string): number => new Date(date).getDate()

export const getMonth = (date: string): number => new Date(date).getMonth()
