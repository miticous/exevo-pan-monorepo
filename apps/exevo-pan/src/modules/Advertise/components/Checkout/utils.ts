import { endpoints } from 'Constants'

const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
export const validateEmail = (email: string): boolean => emailRegex.test(email)

const buildUrl = (nickname: string): string =>
  `${endpoints.TIBIADATA}/${nickname}.json`

export const validateCharacter = async (nickname: string): Promise<boolean> => {
  try {
    const response = await fetch(buildUrl(nickname))
    const data = await response.json()
    return !data.characters.error
  } catch (error) {
    return true
  }
}

export const randomCharacter = (): string => {
  const characters = [
    'Eternal Oblivion',
    'Cachero',
    'Bubble',
    'Setzer Gambler',
    'Arieswar',
    'Kharsek',
    "Lord'Paulistinha",
    'â€ŽMateusz Dragon Wielki',
  ]

  return characters[Math.floor(Math.random() * characters.length)]
}
