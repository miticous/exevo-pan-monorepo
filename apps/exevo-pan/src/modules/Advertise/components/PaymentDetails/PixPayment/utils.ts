import { QrCodePix } from 'qrcode-pix'
import { advertising } from 'Constants'

interface PixParams {
  txId: number
  daysAmount: number
}

export const generateQrCode = async ({
  txId,
  daysAmount,
}: PixParams): Promise<PixObject> => {
  const id = txId.toString()
  const tx = QrCodePix({
    version: '01',
    key: advertising.PIX_KEY,
    name: id,
    city: id,
    message: id,
    transactionId: id,
    value: daysAmount * advertising.BRL_ADVERTISE,
  })

  return {
    payload: tx.payload(),
    qrCode: await tx.base64(),
  }
}
