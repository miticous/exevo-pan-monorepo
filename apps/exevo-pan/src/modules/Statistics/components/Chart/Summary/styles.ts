import styled from 'styled-components'
import { Smooth } from 'styles'
import TrendingIconSvg from 'assets/svgs/trending.svg'
import { PositiveStyleProps } from './types'

export const Wrapper = styled.div`
  * {
    ${Smooth}
  }
`

export const Title = styled.h4`
  font-size: 14px;
  font-weight: 300;
  color: var(--onSurface);
`

export const Value = styled.span`
  margin: 6px 0 4px 0;
  display: flex;
  align-items: center;
  font-size: 20px;
  font-weight: 600;
  color: var(--onSurface);
`

export const TrendIcon = styled(TrendingIconSvg)`
  margin-right: 4px;
  width: 16px;
  height: 16px;
`

export const Percentage = styled.span<PositiveStyleProps>`
  display: flex;
  align-items: center;

  font-size: 12px;
  font-weight: 600;
  color: ${({ positive }) => (positive ? 'var(--green)' : 'var(--red)')};

  ${TrendIcon} {
    fill: ${({ positive }) => (positive ? 'var(--green)' : 'var(--red)')};
    transform: ${({ positive }) => (positive ? 'none' : 'scaleY(-1)')};
  }
`
