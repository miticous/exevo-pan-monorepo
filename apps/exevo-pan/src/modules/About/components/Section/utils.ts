import { copyToClipboard } from 'utils'

export const CopyToClipboard = (id: string): void =>
  copyToClipboard(
    `https://${window.location.hostname}${window.location.pathname}#${id}`,
  )
