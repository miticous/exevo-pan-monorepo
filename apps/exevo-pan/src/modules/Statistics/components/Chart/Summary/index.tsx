import { useTranslations } from 'contexts/useTranslation'
import { memo } from 'react'
import { formatNumberWithCommas } from 'utils/formatNumberWithCommas'
import * as S from './styles'
import { SummaryProps } from './types'

const Summary = ({
  title,
  value,
  percentage,
  positive = true,
  ...props
}: SummaryProps): JSX.Element => {
  const {
    translations: { statistics },
  } = useTranslations()

  return (
    <S.Wrapper {...props}>
      <S.Title>{title}</S.Title>
      <S.Value>{`${formatNumberWithCommas(value)} TC`}</S.Value>
      <S.Percentage positive={positive}>
        <S.TrendIcon
          aria-label={
            statistics.Summary[
              positive ? 'positiveTrendLabel' : 'negativeTrendLabel'
            ]
          }
        />
        {`${percentage.toFixed(2)}%`}
      </S.Percentage>
    </S.Wrapper>
  )
}

export default memo(Summary)
