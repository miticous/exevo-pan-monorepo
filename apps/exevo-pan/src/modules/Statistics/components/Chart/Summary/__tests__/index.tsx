import { screen } from '@testing-library/react'
import { renderWithProviders } from 'utils/test'
import Summary from '..'

describe('<Summary />', () => {
  test('should render values correctly', () => {
    renderWithProviders(
      <Summary title="Total volume" value={1000} percentage={50} />,
    )

    expect(
      screen.getByRole('heading', {
        name: 'Total volume',
      }),
    ).toBeInTheDocument()

    expect(screen.getByText('1,000 TC')).toBeInTheDocument()
    expect(screen.getByText(/50\.00%/)).toBeInTheDocument()
  })

  test('should style trend red/green', () => {
    const { rerender } = renderWithProviders(
      <Summary title="Total volume" value={1000} percentage={50} />,
    )

    const textElement = screen.getByText(/50\.00%/)

    /* expect(textElement).toHaveStyle('color: #377712') */
    expect(textElement).toMatchSnapshot()
    /* expect(screen.getByLabelText('Upwards trend')).toHaveStyle('fill: #377712') */
    expect(screen.getByLabelText('Upwards trend')).toMatchInlineSnapshot(`
      <image
        aria-label="Upwards trend"
        class="styles__TrendIcon-sc-zf1m03-3 efoYRJ"
      />
    `)

    rerender(
      <Summary
        title="Total volume"
        value={1000}
        percentage={50}
        positive={false}
      />,
    )

    /* expect(textElement).toHaveStyle('color: #C51313') */
    expect(textElement).toMatchSnapshot()
    /* expect(screen.getByLabelText('Downwards trend')).toHaveStyle(
      'fill: #C51313',
    ) */
    expect(screen.getByLabelText('Downwards trend')).toMatchSnapshot()

    rerender(
      <Summary title="Total volume" value={1000} percentage={50} positive />,
    )

    /* expect(textElement).toHaveStyle('color: #377712') */
    expect(textElement).toMatchSnapshot()
    /* expect(screen.getByLabelText('Upwards trend')).toHaveStyle('fill: #377712') */
    expect(screen.getByLabelText('Upwards trend')).toMatchSnapshot()
  })
})
