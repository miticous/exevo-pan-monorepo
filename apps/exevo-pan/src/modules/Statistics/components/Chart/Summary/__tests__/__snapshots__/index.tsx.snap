// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> should style trend red/green 1`] = `
<span
  class="styles__Percentage-sc-5rbl8j-4 drShaa"
>
  <image
    aria-label="Upwards trend"
    class="styles__TrendIcon-sc-5rbl8j-3 bRQjW"
  />
  50.00%
</span>
`;

exports[`<Summary /> should style trend red/green 3`] = `
<span
  class="styles__Percentage-sc-5rbl8j-4 fiolhQ"
>
  <image
    aria-label="Downwards trend"
    class="styles__TrendIcon-sc-5rbl8j-3 bRQjW"
  />
  50.00%
</span>
`;

exports[`<Summary /> should style trend red/green 4`] = `
<image
  aria-label="Downwards trend"
  class="styles__TrendIcon-sc-5rbl8j-3 bRQjW"
/>
`;

exports[`<Summary /> should style trend red/green 5`] = `
<span
  class="styles__Percentage-sc-5rbl8j-4 drShaa"
>
  <image
    aria-label="Upwards trend"
    class="styles__TrendIcon-sc-5rbl8j-3 bRQjW"
  />
  50.00%
</span>
`;

exports[`<Summary /> should style trend red/green 6`] = `
<image
  aria-label="Upwards trend"
  class="styles__TrendIcon-sc-5rbl8j-3 bRQjW"
/>
`;
